test/e2e/operator/nmstate_install_test.go

```diff
@@ -130,6 +131,41 @@ var _ = Describe("NMState operator", func() {
 			})
 		})
 	})
+	Context("when cluser-reader exists", func() {
+		const (
+			clusterReaderRoleName = "cluster-reader"
+			testUserNamespace     = "default"
+			serviceAccountName    = "test-user"
+			testUserBindingName   = "test-user-binding"
+		)
+
+		BeforeEach(func() {
+			Expect(createClusterReaderCR(clusterReaderRoleName)).To(Succeed(),
+				"should success creating a clusterrole")
+			Expect(createTestUserSA(testUserNamespace, serviceAccountName)).To(Succeed(),
+				"should success creating a serviceaccount")
+			Expect(createTestUserCRB(testUserBindingName, testUserNamespace, serviceAccountName, clusterReaderRoleName)).To(Succeed(),
+				"should success creating a clusterrolebinding")
+
+			By("Install NMState for the first time")
+			installNMState(defaultOperator.nmstate)
+			eventuallyOperandIsReady(defaultOperator)
+		})
+		AfterEach(func() {
+			uninstallNMStateAndWaitForDeletion(defaultOperator)
+
+			Expect(deleteTestUserCRB(testUserBindingName)).To(Succeed())
+			Expect(deleteTestUserSA(testUserNamespace, serviceAccountName)).To(Succeed())
+			Expect(deleteClusterReaderCR(clusterReaderRoleName)).To(Succeed())
```

Comment:
only delete, if **you** created it

Reply:
all of them ?
We are safe to do so only for the `deleteClusterReaderCR` right ?
since as you said above i will check if its not already exists, and if it exists i won't create nor delete
The two others we know I created because if they are already there, it would fail on the creation.

In case someone interrupts after creation and before deletion, it is better that we will clean those on the run so the next run will be clean.
Their name is unique (has test in it), so it is unlikely someone else created them.

Another point Is that we can remove split those 3 lines to 3 AfterEach so if one fails, it will still perform the others.

wdyt ?

Reply:
> To be more unique, you could also use the `metadata.GenerateName` and let k8s decide on its "final" name.

I prefer deterministic, unless you think its necessary